core/src/test/java/nl/nn/adapterframework/doc/model/AncestorKindsTest.java

```diff
@@ -0,0 +1,100 @@
+package nl.nn.adapterframework.doc.model;
+
+import static nl.nn.adapterframework.doc.model.ElementChild.ALL;
+import static nl.nn.adapterframework.doc.model.ElementChild.IN_XSD;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Class {@link FrankElement} has many different method to get children
+ * and to search ancestors that have children. The tests in this class
+ * are to test all these methods.
+ * @author martijn
+ *
+ */
+public class AncestorKindsTest {
+	private static final String PACKAGE = "nl.nn.adapterframework.doc.testtarget.sparse.";
+
+	private FrankDocModel model;
+
+	@Before
+	public void setUp() {
+		model = FrankDocModel.populate("doc/sparse-digester-rules.xml", PACKAGE + "ContainerChild");
+	}
+
+	@Test
+	public void testConfigChildrenOfPackageSparse() {
+		ConfigChild child = model.findFrankElement(PACKAGE + "ContainerChild")
+				.getConfigChildren(ALL).get(0);
+		assertFalse(child.isDeprecated());
+		child = model.findFrankElement(PACKAGE + "ContainerNoAncestorBecauseChildrenDeprecated")
+				.getConfigChildren(ALL).get(0);
+		assertTrue(child.isDeprecated());
+		child = model.findFrankElement(PACKAGE + "ContainerAncestor")
+				.getConfigChildren(ALL).get(0);
+		assertFalse(child.isDeprecated());
+		child = model.findFrankElement(PACKAGE + "GrandParentWithDeprecatedConfigChildren")
+				.getConfigChildren(ALL).get(0);
+		assertTrue(child.isDeprecated());
+		assertEquals(0, model.findFrankElement(PACKAGE + "AttributeChild")
+				.getConfigChildren(ALL).size());
+		assertEquals(0, model.findFrankElement(PACKAGE + "AttributeNoAncestorBecauseAttributesDeprecated")
+				.getConfigChildren(ALL).size());
+		assertEquals(0, model.findFrankElement(PACKAGE + "AttributeAncestor")
+				.getConfigChildren(ALL).size());
+		assertEquals(0, model.findFrankElement(PACKAGE + "GrandParentWithDeprecatedAttributes")
+				.getConfigChildren(ALL).size());
```

Comment:
Probeer statements zo veel mogelijk op 1 regel te plaatsen. Dat is veel leesbaarder!

Reply:
Gedaan. Is het nu goed?